### DISCUSSION TITLE
No `Unhealthy` detail

### DISCUSSION THREAD

**Discussion by JounQin** [coder/coder#10836] [Announcements]
<img width="1366" alt="image" src="https://github.com/coder/coder/assets/8336744/d2d25d72-c890-4144-8188-2ceade78bdd5">

What url is `curl` requesting that causes the error warning? Is that cause `Unhealthy` status?

**Comment by sharkymark** [comment]
It's curling the code-server IDE

The other curl which appears to be working is installing the agent but that's before the startup script

**Comment by JounQin** [comment]
I want to use code-server IDE, no error details provided, I don't know how to fix it.

**Comment by JounQin** [comment]
I have a code-server IDE running now (install via Homebrew and start service manually, running on port 8080), how can I enable the code-server button here? (How to connect the two together?)

**Comment by JounQin** [comment]
It seems `code-server` is not running correctly, not service on port `13337`:

```log
netstat -lntp G 13337
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)
```

**Comment by matifali** [comment]
If you are only interested in getting a Coder app for code-server. try https://registry.coder.com/modules/code-server

**Comment by JounQin** [comment]
I've already setup `code-server` independently successfully, and I want to try `coder` at the same time.

My problem here is that `code-server` bundlerred by `coder` is not running up successfully, and I don't know why and how to fix that issue.

**Comment by matifali** [comment]
@JounQin, you could use the [code-server module](https://registry.coder.com/modules/code-server) in any of the templates to add it to your workspaces. Coder works by specifying templates and then creating workspaces using them. You need to edit your template and remove the code-server from the `startup_script` in `coder_agent` and instead use the module. Let me know if it helps.

**Comment by JounQin** [comment]
Interesting, after removing the `startup_script` section and rerun `Build template`, the built-in `code-server` in running correctly now even without restarting the workspace... No idea what's going on...

**Comment by JounQin** [comment]
@matifali And I'm not for sure to understand how to use the module

```
module "code-server" {
    source = "https://registry.coder.com/modules/code-server"
    agent_id = coder_agent.example.id
}
```

I've already have a independent `code-server` instance on port `8080`, is that possible to use this instance in `coder` instead?

**Reply by matifali** [reply]
I would ask you to go through coder docs specifically https://coder.com/docs/v2/latest/templates to understand how templates work with Coder.
